packages/aws-cdk-lib/aws-ecs/test/amis.test.ts

```diff
@@ -24,4 +25,46 @@ describe('amis', () => {
       ([k, v]) => k.startsWith(ssmKey) && (v as any).Default.includes(`/bottlerocket/${variant}/x86_64/`),
     )).toEqual(true);
   });
+
+  describe('Amazon Linux 2', () => {
+
+    test('Amazon Linux 2 with kernel', () => {
+      // GIVEN
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app);
+
+      // WHEN
+      ecs.EcsOptimizedImage.amazonLinux2(ecs.AmiHardwareType.STANDARD, { kernel: ec2.AmazonLinux2Kernel.KERNEL_5_10 }).getImage(stack);
+
+      // THEN
+      const assembly = app.synth();
+      const parameters = assembly.getStackByName(stack.stackName).template.Parameters;
+      expect(Object.entries(parameters).some(
+        ([k, v]) => k.startsWith('SsmParameterValueawsserviceecsoptimizedamiamazonlinux2kernel510') && (v as any).Default.includes('/aws/service/ecs/optimized-ami/amazon-linux-2/kernel-5.10/recommended/image_id'),
+      )).toEqual(true);
+
+      expect(Object.entries(parameters).some(
+        ([k, v]) => k.startsWith('SsmParameterValueawsserviceecsoptimizedamiamazonlinux2kernel510') && (v as any).Default.includes(
+          ec2.AmazonLinux2Kernel.KERNEL_5_10.toString(),
+        ),
+      )).toEqual(true);
+    });
+
+    test('Amazon Linux 2 without kernel', () => {
+      // GIVEN
+      const app = new cdk.App();
+      const stack = new cdk.Stack(app);
+
+      // WHEN
+      ecs.EcsOptimizedImage.amazonLinux2(ecs.AmiHardwareType.STANDARD).getImage(stack);
+
+      // THEN
+      const assembly = app.synth();
+      const parameters = assembly.getStackByName(stack.stackName).template.Parameters;
+      expect(Object.entries(parameters).some(
+        ([k, v]) => k.startsWith('SsmParameterValueawsserviceecsoptimizedamiamazonlinux2') && (v as any).Default.includes('/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id'),
+      )).toEqual(true);
+    });
+  });
+
```

Comment:
Can you please add coverage for the `Amazon Linux 2` validation?

Reply:
@lpizzinidev Can you explain more?

Do you mean adding tests that should throw errors when specifying the kernel but the AMI is not Amazon Linux 2?
If so, then I think that won't be necessary because TypeScript will throw a compile error as the `kernel` property is defined only in `EcsOptimizedImage.amazonLinux2`'s arguments.